lib/services/saml.go

```diff
@@ -220,6 +220,10 @@ func CheckSAMLEntityDescriptor(entityDescriptor string) ([]*x509.Certificate, er
 		return nil, trace.Wrap(err, "failed to parse entity_descriptor")
 	}
 
+	if metadata.IDPSSODescriptor == nil {
+		return nil, nil
+	}
+
```

Comment:
Do we want to error on an invalid EntityDescriptor instead of silently swallowing the error? Is this even a valid ED? Would it be more user-friendly to fail fast and tell them their ED is wrong?

Reply:
I have the same question. The godoc even mentions "has at least one cert" which definitely isn't true in this case.

Reply:
You are right, but see my comments from Slack:

> The godoc for this function is incorrect, as clearly there is no check for at least one root to be present, and I'd rather not change that without some additional testing. I think the worst thing that will happen when the roots are missing is merely a failure a little further down the line, so we likely don't even need that check to be present.